src/app/clusters/water-heater-management-server/water-heater-management-server.cpp

```diff
@@ -0,0 +1,196 @@
+{
+    switch (aPath.mAttributeId)
+    {
+    case HeaterTypes::Id:
+        return aEncoder.Encode(mDelegate.GetHeaterTypes());
+    case HeatDemand::Id:
+        return aEncoder.Encode(mDelegate.GetHeatDemand());
+    case TankVolume::Id:
+        if (!HasFeature(Feature::kEnergyManagement))
+        {
+            return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute);
+        }
+        return aEncoder.Encode(mDelegate.GetTankVolume());
+    case EstimatedHeatRequired::Id:
+        if (!HasFeature(Feature::kEnergyManagement))
+        {
+            return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute);
+        }
+        return aEncoder.Encode(mDelegate.GetEstimatedHeatRequired());
+    case TankPercentage::Id:
+        if (!HasFeature(Feature::kTankPercent))
+        {
+            return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute);
+        }
+        return aEncoder.Encode(mDelegate.GetTankPercentage());
+    case BoostState::Id:
+        return aEncoder.Encode(mDelegate.GetBoostState());
+
+    /* FeatureMap - is held locally */
+    case FeatureMap::Id:
+        return aEncoder.Encode(mFeature);
+    }
+
+    /* Allow all other unhandled attributes to fall through to Ember */
+    return CHIP_NO_ERROR;
+}
+
+// CommandHandlerInterface
+CHIP_ERROR Instance::EnumerateAcceptedCommands(const ConcreteClusterPath & cluster, CommandIdCallback callback, void * context)
+{
+    using namespace Commands;
+
+    VerifyOrExit(callback(Boost::Id, context) == Loop::Continue, /**/);
+    VerifyOrExit(callback(CancelBoost::Id, context) == Loop::Continue, /**/);
+
+exit:
+    return CHIP_NO_ERROR;
+}
+
+void Instance::InvokeCommand(HandlerContext & handlerContext)
+{
+    using namespace Commands;
+
+    switch (handlerContext.mRequestPath.mCommandId)
+    {
+    case Boost::Id:
+        HandleCommand<Boost::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleBoost(ctx, commandData); });
+        return;
+    case CancelBoost::Id:
+        HandleCommand<CancelBoost::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleCancelBoost(ctx, commandData); });
+        return;
+    }
+}
+
+void Instance::HandleBoost(HandlerContext & ctx, const Commands::Boost::DecodableType & commandData)
+{
+    uint32_t duration                        = commandData.duration;
+    Optional<bool> oneShot                   = commandData.oneShot;
+    Optional<bool> emergencyBoost            = commandData.emergencyBoost;
+    Optional<int16_t> temporarySetpoint      = commandData.temporarySetpoint;
+    Optional<chip::Percent> targetPercentage = commandData.targetPercentage;
+    Optional<chip::Percent> targetReheat     = commandData.targetReheat;
+
+    //  Notify the appliance if the appliance hardware cannot be adjusted, then return Failure
+    if (!HasFeature(WaterHeaterManagement::Feature::kTankPercent))
+    {
+        if (targetPercentage.HasValue())
```

Comment:
Then this is a spec violation, no?  Doesn't matter whether it's > 100 or not; spec conformance says `[TP]` so if no feature but have value that would be an InvalidCommand status.

Reply:
Fixed. This actually highlighted a problem with the tests for which I will raise a separate change on the test spec.